libs/langchain/langchain/vectorstores/chroma.py

```diff
@@ -474,6 +474,93 @@ def max_marginal_relevance_search(
         )
         return docs
 
+    def max_marginal_relevance_search_by_vector_with_scores(
```

Comment:
can we update max_marginal_relevance_search_by_vector to call this

Reply:
Thank you for the suggestion! Instead of having max_marginal_relevance_search_by_vector call max_marginal_relevance_search_by_vector_with_scores, which would have added an avoidable loop to strip the scores from the results, I chose to create a shared function to reduce redundant code. This function directly handles the candidates prior to iteration.

Reply:
@baskaryan  I've added the docstring to the new shared function and reformatted the code. Please let me know if there's anything else I can improve. I appreciate the time you've taken to review this!

Reply:
EDIT: I revisited my earlier implementation and realized it wasn't optimal. I've made the necessary adjustments and everything should be in order now. Thanks for your patience.